src/runtime_src/core/edge/user/graph_object.cpp

```diff
@@ -0,0 +1,199 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2024 Advanced Micro Devices, Inc. All rights reserved.
+#include "graph_object.h"
+#include "core/edge/user/aie/graph.h"
+#include "core/common/system.h"
+#include <memory>
+
+namespace zynqaie {
+  graph_object::graph_object(ZYNQ::shim* shim, const xrt::uuid& uuid , const char* name,
+                  xrt::graph::access_mode am, const zynqaie::hwctx_object* hwctx)
+    : m_shim{shim}
+  {
+    auto device{xrt_core::get_userpf_device(m_shim)};
+    std::string gname{name};
```

Comment:
Don't think this variable is needed.

Reply:
"gname" is required here as "graph_instance"  constructor is expecting "std:string" , but we have only " char* "

Reply:
Nah, std::string constructs from const char* so you don't need the variable.